src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java

```diff
@@ -661,17 +662,28 @@ public void addEnvVars(List<TemplateEnvVar> envVars) {
         }
     }
 
+    @SuppressWarnings("unused") // Used by jelly
+    @Restricted(DoNotUse.class) // Used by jelly
+    public boolean hasManagePermission() {
+        StaplerRequest2 request = Stapler.getCurrentRequest2();
+        if (request != null) {
+            PodTemplateGroup groupFromRequest = request.findAncestorObject(PodTemplateGroup.class);
+            return groupFromRequest.hasManagePermission();
+        }
+        return Jenkins.get().hasPermission(Jenkins.MANAGE);
+    }
+
     /**
      * Deletes the template.
      */
     @POST
     public HttpResponse doDoDelete(@AncestorInPath PodTemplateGroup owner) throws IOException {
-        Jenkins j = Jenkins.get();
-        j.checkPermission(Jenkins.MANAGE);
         if (owner == null) {
             throw new IllegalStateException("Cloud could not be found");
         }
+        owner.checkManagePermission();
         owner.removeTemplate(this);
```

Comment:
Looks wrong. Check should be done within `PodTemplateGroup` implementation (template methods).